src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -247,26 +291,27 @@ public static ResourceIdentifier of(String service, String instance, String type
     }
 
     private static void checkServiceIsValid(String service) {
```

Comment:
ah right, we probably want something like `#checkServiceMatches(String)` and `#checkInstanceMatches(String)` which are instance methods and produce reasonable error messages when there's a mismatch.